File: mobile/android/app/src/main/java/ru/chernov/todolist/mobile/data/TaskDto.kt
package ru.chernov.todolist.mobile.data

data class TaskDto(
    val id: String,
    val title: String,
    val description: String,
    val startTime: String,
    val endTime: String,
    val priority: PriorityDto,
    val category: CategoryDto,
    val completed: Boolean,
    val overdue: Boolean,
    val createdAt: String,
    val updatedAt: String,
    val ownerUserId: String,
    val sortIndex: Int,
    val recurrenceRule: String,
    val reminderOffsetMinutes: Int,
    val recurrenceEnd: String?,
    val deletedAt: String?
)

enum class PriorityDto {
    URGENT,
    IMPORTANT,
    NORMAL
}

enum class CategoryDto {
    WORK,
    HOME,
    STUDY,
    OTHER
}

File: mobile/ios/ToDoListMobile/Models/TaskDTO.swift
import Foundation

enum TaskPriority: String, Codable {
    case urgent = "URGENT"
    case important = "IMPORTANT"
    case normal = "NORMAL"
}

enum TaskCategory: String, Codable {
    case work = "WORK"
    case home = "HOME"
    case study = "STUDY"
    case other = "OTHER"
}

struct TaskDTO: Identifiable, Codable {
    let id: String
    let title: String
    let description: String
    let startTime: String
    let endTime: String
    let priority: TaskPriority
    let category: TaskCategory
    let completed: Bool
    let overdue: Bool
    let createdAt: String
    let updatedAt: String
    let ownerUserId: String
    let sortIndex: Int
    let recurrenceRule: String
    let reminderOffsetMinutes: Int
    let recurrenceEnd: String?
    let deletedAt: String?
}
